app/controllers/projects_controller.rb

```diff
@@ -0,0 +1,11 @@
+class ProjectsController < ApplicationController
+  def show
+    @project = Project.find(params[:id])
+  end
+
+  def update
```

Comment:
So if we are thinking about what is really happening when we read this:
```
When I visit a project's show page
I see a form to add a contestant to this project
When I fill out a field with an existing contestants id
```
We aren't truly updating a project. A RESTful update would be changing some attribute associated with a project. What this is actually saying is hey take an existing contestant and add them to this existing project. So we are essentially creating a new `ContestantProject` object. So the more RESTful way to create this would be to build a `project_contestant_controller` with a `create` action. That would look something like:
```ruby
class ProjectContestantsController < ApplicationController
  def create
    @project = Project.find(params[:project_id])
    @contestant = Contestant.find(params[:contestant_id])
    ContestantProject.create!(project: @project, contestant: @contestant)

    redirect_to "/projects/#{@project.id}"
  end
end
```
and your route would need to point to this. So something like:
```ruby
post "/project/:project_id/contestants", to: "project_contestants#create"
```